src/main/java/com/google/devtools/build/lib/rules/cpp/CppModuleMapAction.java

```diff
@@ -104,117 +98,122 @@ public CppModuleMapAction(
   }
 
   @Override
-  public DeterministicWriter newDeterministicWriter(ActionExecutionContext ctx)  {
+  public boolean makeExecutable() {
+    // In theory, module maps should not be executable but, in practice, we don't care. As
+    // 'executable' is the default (see ActionOutputMetadataStore.setPathReadOnlyAndExecutable()),
+    // we want to avoid the extra file operation of making this file non-executable.
```

Comment:
@justinhorvitz I wonder whether this comment is still correct in light of
https://github.com/bazelbuild/bazel/blob/99963ee16472df059635ea92a215a3306badd015/src/main/java/com/google/devtools/build/lib/exec/FileWriteStrategy.java#L63

It looks like having `makeExecutable()` return `true` always results in that file operation, so not sure whether possibly saving another one elsewhere gains anything. I could drop this if you think it's not needed.

Reply:
I looked up the history of this comment and found more context. It is referring to a google internal implementation of `FileWriteStrategy` which only calls `setExecutable(false)` because our remote execution strategy already makes files executable by default. You can add this clarification to the comment if you wish.

Reply:
I see. Could it make sense to have `makeExecutable` be ternary (with a "don't care" state) to make this moot efficient for Bazel?

Reply:
It's a good idea but I don't know how to measure the value. I also wonder whether we really need to make anything explicitly not executable, or whether false can be used for "don't care."

Reply:
I also don't know how to assess whether this is even worth it. Since `ctx.actions.write` exposes the executable bit in public API, we would probably need ternary state somewhere. I will consider looking into this as a follow-up at some point.